src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -73,50 +82,100 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
         }
 
         public void AddNew() {
-            SelectConnection(new ConnectionViewModel());
+            _shell.AssertIsOnMainThread();
+            CancelEdit(SelectedConnection);
+            NewConnection = new ConnectionViewModel();
+        }
+        
+        public void CancelEdit(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            if (connection == NewConnection) {
+                NewConnection = null;
+            } else if (connection != null) {
+                connection.Reset();
+                connection.IsEditing = false;
+            }
+        }
+
+        public void BrowseLocalPath(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            string latestLocalPath;
+            Uri latestLocalPathUri;
+
+            if (connection.Path != null && Uri.TryCreate(connection.Path, UriKind.Absolute, out latestLocalPathUri) && latestLocalPathUri.IsFile && !latestLocalPathUri.IsUnc) {
+                latestLocalPath = latestLocalPathUri.LocalPath;
+            } else { 
+                latestLocalPath = Environment.SystemDirectory;
+
+                try {
+                    latestLocalPath = new RInstallation().GetCompatibleEnginePathFromRegistry();
+                    if (string.IsNullOrEmpty(latestLocalPath) || !Directory.Exists(latestLocalPath)) {
+                        // Force 64-bit PF
+                        latestLocalPath = Environment.GetEnvironmentVariable("ProgramFiles");
+                    }
+                }
+                catch (ArgumentException) { }
+                catch (IOException) { }
+            }
+
+            var path = _shell.ShowBrowseDirectoryDialog(latestLocalPath);
+            if (path != null) {
+                connection.Path = path;
+            }
         }
 
-        public void CancelSelected() {
+        public void Edit(IConnectionViewModel connection) {
             _shell.AssertIsOnMainThread();
-            SelectedConnection?.Reset();
+            CancelEdit(NewConnection);
+            CancelEdit(SelectedConnection);
+            SelectedConnection = connection;
+            connection.IsEditing = true;
         }
 
-        public void SaveSelected() {
+        public Task TestConnectionAsync(IConnectionViewModel connection) => Task.CompletedTask;
+
+        public void Save(IConnectionViewModel connectionViewModel) {
             _shell.AssertIsOnMainThread();
-            if (string.IsNullOrEmpty(SelectedConnection.Name)) {
+            if (string.IsNullOrEmpty(connectionViewModel.Name)) {
```

Comment:
Yes, it is disabled already. 
Moved strings to the save button tooltip

---
In reply to: [77056353](https://github.com/Microsoft/RTVS/pull/2237#discussion_r77056353) [](ancestors = 77056353)